### PR TITLE
Fix nullability of non-optional fields in TS interfaces and class-level JS comments

### PR DESCRIPTION
**Pull request by martin-traverse**

### CODE REVIEW

**Comment by martin-traverse** [comment]
Relevant issue is #1779

**Comment by mamiksik** [comment]
Hey, would you have ETA for when this will be merged please?

**Comment by andrea11** [comment]
Hello, any chance to see this merged?

**Comment by martin-traverse** [comment]
Hello - I have rebased this PR on top of the latest master. Also as per #1779, the change is hidden behind a new flag --pb3-optional. If the flag is not specified, the old behaviour is retained.

@alexander-fenster - are you happy to merge with these changes? Let me know if there is anything else you think would be helpful. We're coming up to the next release series of our product, so for us it would be great to have this in time for that release.

File: cli/pbjs.js [hunk 1]
```diff
@@ -148,6 +149,7 @@ exports.main = function main(args, callback) {
                 "  --force-message  Enforces the use of message instances instead of plain objects.",
                 "",
                 "  --null-defaults  Default value for optional fields is null instead of zero value.",
+                "  --pb3-optional   Make type declarations respect optional fields for PB3.",
```
Comment (Asondo) [inline cli/pbjs.js:152]: May be just "--disable-defaults" and thats all? There is no data in grpc package on default value, so, why we should have some magic default or unexpected null, it is undefined and thats all. So, just disable defaults and it'll be ok)

**Comment by martin-traverse** [comment]
I have rebased this change on top of the latest master. @alexander-fenster are you ok to review / approve this change now? Is there anything else you need from me?

**Comment by martin-traverse** [comment]
Hello - I am closing this PR as I have a substantially new implementation

**Comment by martin-traverse** [comment]
Please see #2011 for a more complete solution to this issue.

**Pull request closed without merging**